server/game/core/gameSystem/CardTargetSystem.ts

```diff
@@ -142,28 +142,24 @@ export abstract class CardTargetSystem<TContext extends AbilityContext = Ability
     }
 
     public override checkEventCondition(event: any, additionalProperties = {}): boolean {
-        return this.canAffect(event.card, event.context, additionalProperties);
+        return this.canAffect(event.card, event.context, additionalProperties, true);
     }
 
-    public override canAffect(card: Card, context: TContext, additionalProperties = {}): boolean {
+    public override canAffect(card: Card, context: TContext, additionalProperties: any = {}, mustChangeGameState = false): boolean {
```

Comment:
NIT: Record<string, any> would be better here
```suggestion
    public override canAffect(card: Card, context: TContext, additionalProperties: Record<string, any> = {}, mustChangeGameState = false): boolean {
```

Reply:
Agreed, but I would prefer to change the type for that property everywhere all at once instead of piecemeal. I have a few noisy changes that are due, I might batch them all together and do them in one loud PR sometime soon just to get them out of the way.